### ISSUE TITLE
Use $HOME/.config/Microsoft/ASP.NET instead of $HOME/.microsoft/usersecrets/ in unix-like environments.

### ISSUE THREAD

**Issue by ariellourenco** [dotnet/runtime#101012]
### Description

This is a classic issue raised in any project. Currently .NET runtime pollutes user `$HOME` directory by placing an unnecessary folder there:

```cs
// For backwards compat, this checks env vars first before using Env.GetFolderPath
 string? appData = Environment.GetEnvironmentVariable("APPDATA");
 string? root = appData                                                                   // On Windows it goes to %APPDATA%\Microsoft\UserSecrets\
    ?? Environment.GetEnvironmentVariable("HOME")                 // On Mac/Linux it goes to ~/.microsoft/usersecrets/
    ?? Environment.GetFolderPath(Environment.SpecialFolder.ApplicationData)
    ?? Environment.GetFolderPath(Environment.SpecialFolder.UserProfile)
    ?? Environment.GetEnvironmentVariable(userSecretsFallbackDir);           
```

This is a common pain for a long time for many people on unix-like platforms. Interestingly, on Windows, the runtime is a good citizen and follow Microsoft recommendations for folder locations, storing user secrets under `APPDATA`. However, I cannot say the same for unix-like platforms. 

### Reproduction Steps

To reproduces the problem run the `init` command for the  Secret Manager tool in the project directory.

### Expected behavior

Files stored by .NET runtime should follow the [XDG Base Directory specification](https://specifications.freedesktop.org/basedir-spec/latest/ar01s03.html) and place user configuration files in `$XDG_DATA_HOME/` (default ~/.local/share). 

On Mac, we should use `~/Library/Application Support`

### Actual behavior

As seen on Linux and Mac, .NET runtime leave config files in `~/.microsoft/usersecrets`

### Other information

There are a few issues related to this one across Microsoft repos.

- [https://github.com/dotnet/aspnetcore/issues/43278](https://github.com/dotnet/aspnetcore/issues/43278)
- [https://github.com/dotnet/sdk/issues/8678](https://github.com/dotnet/sdk/issues/8678)
- [https://github.com/microsoft/vscode-dotnettools/issues/514](https://github.com/microsoft/vscode-dotnettools/issues/514)

**Comment by jkotas** [comment]
Offending code: https://github.com/dotnet/runtime/blob/2e8f818677e55ef7570b9cd3b21890fa9ac93b0f/src/libraries/Microsoft.Extensions.Configuration.UserSecrets/src/PathHelper.cs#L59-L67

**Comment by artl93** [comment]
@jkotas  - handing off to copilot: I am running an experiment on nominating issues for copilot actionability.

**Comment by artl93** [comment]
Sending back to further triage. We have a PR, but taking this would create problems beyond just being a breaking change: 
- Version to version upgrades would "lose" already set secrets in their environments (dev or otherwise) 
- There may be any variety of tools that use these locations as baseline assumptions for secrets

Based on these points, I wonder if we should let this go. Yeah, it's inconsistent with the platform, but I wonder if it's worth it. Even making a fallback backward compatible seems like a bad idea in my head. Thoughts?

**Comment by ariellourenco** [comment]
@artl93,

Both are legitimate risks, and we want to avoid breaking existing workflows. At the same time, the community has repeatedly expressed frustration with `$HOME` becoming cluttered with configuration files, and moving the secret store to a dedicated location would be a welcome improvement for many users.

**Proposed migration strategy**

- Detect an existing legacy folder ($HOME/.microsoft/usersecrets, for example).
   If it exists, we move its contents to the new location.
   Then we create a symlink from the old path to the new one. This preserves the original path for any tooling that still expects it, while allowing the application to read/write from the canonical location.
- Fresh installations (no legacy folder detected) simply create the directory in the new location without any symlink.
- Upgrade path – on subsequent runs we can check whether the symlink is still present. If the user explicitly opts out (e.g., via a config flag), we can remove the symlink and migrate everything permanently to the new location.

**Comment by tarekgh** [comment]
I don’t think we need to make any complex changes for that. If we really want to fix it, we should follow the approach in the linked PR, documenting the breaking change and possibly adding a configuration switch to allow reverting to the previous behavior.
